lib/ui/painting/image_generator_apng.cc

```diff
@@ -536,6 +536,13 @@ bool APNGImageGenerator::DemuxNextImageInternal() {
           SkCodecAnimation::DisposalMethod::kKeep) {
     // Mark the required frame as the previous frame in all cases.
     image->frame_info->required_frame = images_.size() - 1;
+  } else if (images_.size() > (first_frame_index_ + 1) &&
+             images_.back().frame_info->disposal_method ==
+                 SkCodecAnimation::DisposalMethod::kRestorePrevious) {
```

Comment:
There's a debug log above this code (line 527) that should be removed now that we support `kRestorePrevious`.